### PR TITLE
Added the makrdup step in the RNAseq pipeline

### PR DESCRIPTION
**Pull request by saimukund20**

### CODE REVIEW

File: definitions/pipelines/rnaseq.cwl [hunk 1]
```diff
@@ -52,8 +52,11 @@ inputs:
 outputs:
     final_bam:
         type: File
-        outputSource: index_bam/indexed_bam
+        outputSource: mark_dup/sorted_bam
         secondaryFiles: [.bai]
+    markdup_metrics:
```
Comment (chrisamiller) [inline definitions/pipelines/rnaseq.cwl]: I guess it doesn't hurt to output these metrics, but it's probably not really needed!

File: definitions/pipelines/rnaseq.cwl [hunk 2]
```diff
@@ -129,10 +132,16 @@ steps:
             bam: merge/merged_bam
         out:
             [indexed_bam]
+    mark_dup:
```
Comment (chrisamiller) [inline definitions/pipelines/rnaseq.cwl]: We don't need to output a separate markdup bam - just need to make sure the final bam is derived from it (as you did above!)

File: definitions/pipelines/rnaseq.cwl [hunk 3]
```diff
@@ -129,10 +132,16 @@ steps:
...
+        run: ../tools/mark_duplicates_and_sort.cwl
+        in:
+            bam: merge/merged_bam
+        out:
+            [sorted_bam, metrics_file]
     stringtie:
         run: ../tools/stringtie.cwl
         in:
-            bam: merge/merged_bam
+            bam: mark_dup/sorted_bam
```
Comment (chrisamiller) [inline definitions/pipelines/rnaseq.cwl]: Did you verify that stringtie ignores duplicate marking?  (either in their docs/online or by testing it yourself?)

File: definitions/pipelines/rnaseq.cwl (hunk 2 above)
Comment (chrisamiller) [reply definitions/pipelines/rnaseq.cwl]: my bad - I misread this. it's fine!

File: definitions/pipelines/rnaseq.cwl (hunk 1 above)
Comment (saimukund20) [reply definitions/pipelines/rnaseq.cwl]: removed that as an output.

File: definitions/pipelines/rnaseq.cwl (hunk 3 above)
Comment (saimukund20) [reply definitions/pipelines/rnaseq.cwl]: I couldn't find any literature of stringtie that says it ignores duplicate marking, but i ran a test and the outputs with/without markdup were the same.

**Pull request merged**